webpack.config.ts

```diff
@@ -232,63 +221,110 @@ if (EEGVisEnabled !== 'true' && EEGVisEnabled !== '1' ) {
+  const allModules: Record<string, string[]> = {};
+  for (const [moduleName, moduleEntryPoints] of
+    Object.entries(projectModules)
+  ) {
+    allModules[moduleName] = [...moduleEntryPoints];
+  }
+
+  // Add project-specific modules and overrides to the record of modules
+  for (const [moduleName, moduleEntryPoints] of
+    Object.entries(projectModules)
+  ) {
+    if (moduleName in allModules) {
+      allModules[moduleName].push(...moduleEntryPoints);
     } else {
-      lorisModules[moduleName] = files;
+      allModules[moduleName] = moduleEntryPoints;
     }
   }
+
+  return allModules;
 }
 
-// Only build the given target module if there is one in the current
-// environment.
-const target = process.env.target;
-if (target) {
-  if (target in lorisModules) {
-    lorisModules = {
-      [target]: lorisModules[target],
-    };
-  } else {
-    console.error(`Target module '${target}' not found.`);
+/**
+ * Filter the record of LORIS modules to contain only the target module if a
+ * target is defined, or return the record unchanged if no target is defined.
+ */
+function filterTargetModules(
+  modules: Record<string, string[]>,
+): Record<string, string[]> {
+  // If there is no target module, do not filter the modules
+  if (!target) {
+    // eslint-disable-next-line no-console
+    console.log('Building all modules');
+    return modules;
+  }
+
+  // Exit if the target module is not found in the list of modules
+  if (!(target in modules)) {
+    console.error(`Target module \''${target}'\' not found`);
     process.exit(1);
   }
+
+  // Return a record containing only the target module files
+  // eslint-disable-next-line no-console
+  console.log(`Building module \'${target}\'`);
+  return {
+    [target]: modules[target],
+  };
 }
 
-// Transform the mapping of LORIS modules to a mapping of webpack entry points.
-const entries = Object.fromEntries(
-  Object.entries(lorisModules)
-    .map(([name, files]) => makeModuleEntries(name, files))
-    .flat()
-);
+/**
+ * Get the Webpack entries of a given module, with each entry being mapped to
+ * its name.
+ */
+function getModuleEntries(
+  moduleName: string,
+  moduleEntryPoints: string[],
+): Record<string, webpack.EntryOptions>[] {
+  // Check if a project override exists for the module.
+  const basePath = fs.existsSync(`./project/modules/${moduleName}`)
+    ? `./project/modules/${moduleName}/`
+    : `./modules/${moduleName}/`;
+
+  return moduleEntryPoints.map((moduleEntryPoint) => ({
+    [moduleName + '/' + moduleEntryPoint]: {
+      import: basePath + 'jsx/' + moduleEntryPoint,
+      filename: basePath + 'js/' + moduleEntryPoint + '.js',
+      library: {
+        name: ['lorisjs', moduleName, moduleEntryPoint],
+        type: 'window',
+      },
+    },
+  }));
+}
+
+/**
+ * Get the Webpack entries of the LORIS modules to build, with each entry being
+ * mapped to its name.
+ */
+function getModulesEntries(): Record<string, webpack.EntryOptions> {
```

Comment:
I think the return type of Object.assign is not the same as the return type mentioned up here

Reply:
It is ! `moduleEntries` is of type `Record<string, webpack.EntryOptions>[]`. `Object.assign` is used with the spread of `moduleEntries` to merge all of these objects into a single object. The Typescript compiler does type checking here, so it would have signaled an error if the type did not match.

![image](https://github.com/user-attachments/assets/58c7db08-21e8-4a90-a3ce-ea84bae22229)

Reply:
Hmm, I see that the type of the returned value is `Record<string, webpack.EntryOptions>[]`, but doesn't the return type of the function say `Record<string, webpack.EntryOptions>`? With no `[]`?

Reply:
ohhhhh I see - thanks!